src/turbopuffer/_base_client.py

```diff
@@ -675,10 +675,6 @@ def user_agent(self) -> str:
     def base_url(self) -> URL:
         return self._base_url
 
-    @base_url.setter
-    def base_url(self, url: URL | str) -> None:
-        self._base_url = self._enforce_trailing_slash(url if isinstance(url, URL) else URL(url))
-
```

Comment:
Allowing changing the `base_url` after client construction seems not very important and messes with the region substitution, so I just removed it.